### ISSUE TITLE
Slack "Sign in with Google" not working

### ISSUE THREAD

**Issue by CHarnel** [getferdi/ferdi#1350]
**Bug description:**

On a fresh install, trying to log into slack via "Sign in with Google" and it always goes back to the Sign in to XXXX page. Tried logging in with email, ended up on that same page again. Works in a browser.

**Steps to reproduce:**

1. Install the latest version
2. Add slack service
3. Try to login
4. See the Sign in to [workspace_name] page

**Expected behavior**

Expected it to sign me into slack.

**Screenshots**

![image](https://user-images.githubusercontent.com/9692576/117110529-eababd00-ad8e-11eb-9513-0a30d631b9c3.png)
![image](https://user-images.githubusercontent.com/9692576/117110586-06be5e80-ad8f-11eb-84fe-191fdb69dcf1.png)

**Environment:**

- Operating System: macOS Big Sur (ver 11.2.3)
- Ferdi Version: 5.5.0 (5.5.0.2116)
- Server: Using without an account

- Debug information:
Both Copy and Publish options aren't working

**Comment by vraravam** [comment]
I have tested that I am able to login via google auth into slack.

I am running this version of the Ferdi app:
<img width="280" alt="Screen Shot 2021-05-14 at 2 43 13 PM" src="https://user-images.githubusercontent.com/69629/118249154-b8ae0700-b4c2-11eb-85e6-d4a2f3740cec.png">

This version is the latest from the GH repo, built from source. I have uploaded the version [here](https://drive.google.com/drive/folders/1mJmKbhpVHdb066IHlrdYiD8TG9wlmuBv) - since the nightly build and publish process is currently broken. Please try with this version.

Note: There are separate builds for ARM vs x86 hardware architectures.

**Comment by czettnersandor** [comment]
Same issue here. If I enable the "Navigation bar behaviour" from the settings to show navigation bar on all services, The URL where it's stuck is this:

https://accounts.google.com/o/saml2/idp?idpid=C03grh1hh&SAMLRequest=jVPbjtowEH3fr0...

And there's some flashing in the background. If I re-enter the slack URL to the navigation bar, I'm getting the login screen again.

**Comment by czettnersandor** [comment]
One more thing I must add, that may help: it's happening for me only for Slack Enterprise (URL is ***.enterprise.slack.com) And it's broken even if I add it as a Custom Website. Works from the browser.

**Comment by vraravam** [comment]
@czettnersandor - as I had requested, is it possible for you to try out the updated version of Ferdi? As I have mentioned, I have been using Slack (with google SSO - but, not the enterprise one), and I do not have this issue. Can we please try to eliminate the variables that might cause this incorrect behavior?

**Comment by sad270** [comment]
i have the same issue with
```
Version: 5.6.0-beta.6
Electron: 13.0.1
Chrome: 91.0.4472.69
Node.js: 14.16.0
Platform: macOS Catalina
Arch: x64
Build date: Sun May 30 2021 19:03:02 GMT+0200 (Central European Summer Time)
Git SHA: 48e25d8
Git branch: HEAD
```
And I have 2 factor auth in my google account

### WHat i do to fix this issue
Delete Slack services
Quit Ferdi
Replace the `index.js` of slack recipes (in `/Users/YOUR_USERNAME/Library/Application Support/Ferdi/recipes/slack/index.js`) by
```js
"use strict";

module.exports = (Franz) =>
    class Slack extends Franz {
        overrideUserAgent() {
            return 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36';
        }
        modifyRequestHeaders() {
            return [
                {
                    headers: {
                        'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36',
                    },
                    requestFilters: {
                        urls: ['*://*/*'],
                    }
                }
            ]
        }
    };

```
(This is the current user agent of brave browser if you read this later you can use the latest version UA of your favorite Browser)
Open Ferdi
Create slack services and try to connect again.

When you log in again he should ask your email and password again, you should not have the "Choose an account" page. If you have this page try to Quite ferdi and delete/recreate slack service. You can try to clear Ferdi cache in settings Advanced, but you will lose your settings and you will be disconnected from ferdi account

### Some additional informations about why it's not working
I don't really know but I think the Google 2F Auth is borken in Ferdi when we are in "Sign in with Google"
Because after typing my password i don't have the same 2F Validation page
This 2 screen is at the same step, first is whithout changing anythink in Ferdi (After fresh install) and the second is same step, but after do the fix in this comment.
In the first screen i have to click on blue button to confirm that i click on "Yes" on my phone, and in the second i don't have to, i just click on "Yes" on my phone and he log me in automatically.

### Without changing User Agent
<img width="387" alt="Screenshot 2021-05-31 at 10 34 29" src="https://user-images.githubusercontent.com/5607440/120166804-48f78600-c1fd-11eb-9898-5f51e151fbe9.png">

### With changing User Agent
<img width="472" alt="Screenshot 2021-05-31 at 10 30 30" src="https://user-images.githubusercontent.com/5607440/120167010-85c37d00-c1fd-11eb-9f69-0af4f85dbca4.png">

**Comment by kris7t** [comment]
@sad270 

> (This is the current user agent of brave browser if you read this later you can use the latest version UA of your favorite Browser)

Please only do that if your favorite browser is Chromium-based, otherwise some Google features may not work inside Ferdi (it tries to run code meant for other browsers but fails, since Ferdi runs on Chromium), and might even get your account locked if Google notices the discrepancy.

**Comment by sad270** [comment]
Hello, some updates...
I install last nightly `5.6.0-nightly.71` with the new feature to override user-agent in service, and I change my Slack service user agent by the UA of Brave, try to connect with Google Account and this is not working. So i thinnks it's not only a UA issue.

**Comment by JustoHermano** [comment]
Having this issue on lastest released Version: 5.6.0

**Comment by sad270** [comment]
Fixed in `5.6.3-nightly.15`
To use Google SSO :
1. Open https://whatmyuseragent.com/ with a chromium-based browser (like Google Chrome, Chromium, Brave ...)
2. Copy your user agent (for exemple `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.82 Safari/537.36`)
3. Go to your Slack service settings
4. Past the User agent in the `User Agent`field
5. Save and restart Ferdi.
6. Now you can `Sign in with Google` !
7. Clear the UserAgent field in your slack settings after logging in. (You juste need it for the login)
